src/ProxyHandler.cc

```diff
@@ -3,16 +3,159 @@
 const std::string HOST_TOKEN = "host";
 const std::string PORT_TOKEN = "port";
 
+using boost::asio::ip::tcp;
+
 RequestHandler::Status ProxyHandler::HandleRequest(const Request& request,
                      Response* response) {
-    response->SetStatus(Response::ok);
-    response->SetVersion("HTTP/1.0");
-    response->AddHeader("Content-Type", "text/plain");
-    response->SetBody(request.raw_request());
+
+    // Get endpoints corresponding to the server name.
+    boost::asio::io_service io_service;
```

Comment:
Agreed, or at least having the ProxyHandler have a socket as one of its member variables and having separate send and receive functions for requests.

That'll probably be one of the next things we implement